packages/amplify-graphql-model-transformer/src/resolvers/rds/resolver.ts

```diff
@@ -196,14 +196,28 @@ export const createRdsLambda = (
  * add the name. We can figure out the right way to expose this to customers if needed, but for now we are not invoking `setResourceName`
  * because it would have no effect.
  */
-export const createLayerVersionCustomResource = (scope: Construct, resourceNames: SQLLambdaResourceNames): AwsCustomResource => {
+export const createLayerVersionCustomResource = (
+  scope: Construct,
+  resourceNames: SQLLambdaResourceNames,
+  context: TransformerContextProvider,
+): AwsCustomResource => {
   const { SQLLayerManifestBucket, SQLLayerManifestBucketRegion, SQLLayerVersionManifestKeyPrefix } = ResourceConstants.RESOURCES;
 
   const key = Fn.join('', [SQLLayerVersionManifestKeyPrefix, Fn.ref('AWS::Region')]);
 
   const manifestArn = `arn:aws:s3:::${SQLLayerManifestBucket}/${key}`;
 
   const resourceName = resourceNames.sqlLayerVersionResolverCustomResource;
+
+  // if deployment type is sandbox, use id in the format: resourceName-YYYY-MM-DD to avoid multiple deployments in the same day
+  // if deployment type is branch, use id in the original format
+  let physicalIdValue: string;
+  if (context?.deploymentIdentifier?.deploymentType === 'sandbox') {
+    physicalIdValue = `${resourceName}-${new Date().toISOString().substring(0, 10)}`;
```

Comment:
Why is "per day" the right granularity for controlling deployments in a sandbox? Why would we try to compromise on this at all? By introducing an arbitrary cutoff, we're going to still require deployments for some customers whose sandbox deployments happen to span a day boundary, while missing layer version updates for customers who do multiple deployments that don't happen to span a day boundary. Since sandboxes aren't intended for production, why can't we just use a strictly deterministic resource name, based e.g. on the sandbox name?

Reply:
I was discussing with Christ, Phani, Dane, and Josef, the major concern we have is that, if we use a fixed name like the resourceName, it will not pick up the weekly release/update that might contain security patches. Between weekly and daily, we decided to go with daily. 

I'm still trying to understand the nuances here. Assuming we have a layer version update at noon, and a user deploys once in the morning, and once in the afternoon. 
1) For the current approach using Date.now(), this user will have the update
2) For "per day" approach, this user will not have the update. Unless they deploy again the next day
3) For fixed name approach with only `${resourceName}`, they will never get this update no matter how many times they deploy (i'm not sure if i get it right here)